tpm2/credactivation/credential_activation.go

```diff
@@ -0,0 +1,110 @@
+package credactivation
+
+import (
+	"crypto"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+// Generate returns an idObject & wrapped secret for use in credential activation.
+// This has been tested on EKs compliant with TCG 2.0 EK Credential Profile
+// specification, revision 14.
+// The pub parameter must be a pointer to rsa.PublicKey.
+// The secret parameter must not be longer than the longest digest size implemented
+// by the TPM. A 32 byte secret is a safe, recommended default.
+//
+// This function implements Credential Protection as defined in section 24 of the TPM
+// specification revision 2 part 1, with the additional caveat of not supporting ECC EKs.
+// See: https://trustedcomputinggroup.org/resource/tpm-library-specification/
+func Generate(aik *tpm2.HashValue, pub crypto.PublicKey, symBlockSize int, secret []byte) ([]byte, []byte, error) {
+	rsaPub, ok := pub.(*rsa.PublicKey)
+	if !ok {
+		return nil, nil, errors.New("only RSA public keys are supported for credential activation")
+	}
+
+	return generate(aik, rsaPub, symBlockSize, secret, rand.Reader)
+}
+
+func generate(aik *tpm2.HashValue, pub *rsa.PublicKey, symBlockSize int, secret []byte, rnd io.Reader) ([]byte, []byte, error) {
```

Comment:
`generateRSA`. Then later it'll be easier to add ECC

Reply:
Done.